openapi-integration-tests/src/test/java/io/ballerina/openapi/idl/client/IDLClientGenPluginTests.java

```diff
@@ -0,0 +1,106 @@
+/*
+ * Copyright (c) 2022, WSO2 LLC. (http://www.wso2.com). All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package io.ballerina.openapi.idl.client;
+
+import io.ballerina.openapi.OpenAPITest;
+import io.ballerina.openapi.TestUtil;
+import org.testng.Assert;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Test;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+
+import static io.ballerina.openapi.TestUtil.DISTRIBUTIONS_DIR;
+import static io.ballerina.openapi.TestUtil.RESOURCES_PATH;
+import static io.ballerina.openapi.TestUtil.executeRun;
+import static io.ballerina.openapi.TestUtil.getMatchingFiles;
+
+/**
+ * This class is for client IDL integration tests.
+ */
+public class IDLClientGenPluginTests extends OpenAPITest {
+    public static final String DISTRIBUTION_FILE_NAME = DISTRIBUTIONS_DIR.toString();
+    public static final Path TEST_RESOURCE = Paths.get(RESOURCES_PATH.toString() + "/client-idl-projects");
+
+    @BeforeClass
+    public void setupDistributions() throws IOException {
+        TestUtil.cleanDistribution();
+    }
+
+    @Test
+    public void testValidSwaggerContract() throws IOException, InterruptedException {
+        File[] matchingFiles = getMatchingFiles("project_01");
+        assert matchingFiles != null;
+        Assert.assertEquals(matchingFiles.length, 1);
+    }
```

Comment:
Shouldn't we need to assert file names as well?

Reply:
This is to check whether the directory has been created because generated module contains a random number in containing it and is unable to resolve the path for it.